src/Djot/Blocks.hs

```diff
@@ -1127,12 +1129,12 @@ closeContainingSections lev = do
       closeContainingSections lev
     _ -> pure ()
 
--- TODO avoid detour through String
+-- TODO avoid detour through Text
 toIdentifier :: ByteString -> ByteString
 toIdentifier bs =
   if null parts
      then "sec"
-     else strToUtf8 $ intercalate "-" parts
+     else encodeUtf8 $ T.intercalate (T.singleton '-') parts
  where
    isSym = (`elem` ("][~!@#$%^&*(){}`,.<>\\|=+/" :: [Char]))
-   parts = words $ map (\c -> if isSym c then ' ' else c) $ utf8ToStr bs
+   parts = T.words $ T.map (\c -> if isSym c then ' ' else c) $ decodeUtf8With lenientDecode bs
```

Comment:
I just realised I can also change this to the equivalent [`decodeUtf8Lenient`](https://hackage.haskell.org/package/text-2.1/docs/Data-Text-Encoding.html#v:decodeUtf8Lenient), let me know if you would like me to do that instead.

Reply:
No, because `decodeUtf8Lenient` is relatively recent and some older text versions don't have it.